packages/derive/src/lib.rs

```diff
@@ -55,27 +53,26 @@ use std::str::FromStr;
 pub fn entry_point(_attr: TokenStream, mut item: TokenStream) -> TokenStream {
     let cloned = item.clone();
     let function = parse_macro_input!(cloned as syn::ItemFn);
-    let name = function.sig.ident.to_string();
+
     // The first argument is `deps`, the rest is region pointers
     let args = function.sig.inputs.len() - 1;
+    let fn_name = function.sig.ident;
+    let wasm_export = format_ident!("__wasm_export_{fn_name}");
+    let do_call = format_ident!("do_{fn_name}");
 
-    // E.g. "ptr0: u32, ptr1: u32, ptr2: u32, "
-    let typed_ptrs = (0..args).fold(String::new(), |acc, i| format!("{acc}ptr{i}: u32, "));
-    // E.g. "ptr0, ptr1, ptr2, "
-    let ptrs = (0..args).fold(String::new(), |acc, i| format!("{acc}ptr{i}, "));
+    let decl_args = (0..args).map(|item| format_ident!("ptr_{item}"));
+    let call_args = decl_args.clone();
 
-    let new_code = format!(
-        r##"
+    let new_code = quote! {
```

Comment:
Yeah, `quote` just makes this more maintainable. And it's in the dependency tree already since the `proc-macro` feature (which is needed for the `parse_macro_input!()`) already pulls in `quote` unconditionally anyway.